src/Environments/include/State.tcc

```diff
@@ -18,11 +18,10 @@
 #include <iostream>
 
 template <typename T> T State::getData() const {
-    // TODO: wrap in try-catch
     try {
         return std::any_cast<T>(m_data->getData());
     } catch (const std::bad_any_cast &e) {
-        std::cout << e.what() << '\n';
+        throw;
```

Comment:
Are you sure this preserves stack trace?
```
auto p = std::current_exception();
std::rethrow_exception (p);
```
exists, not sure if it's just a more verbose version

Reply:
It looks like a throw; statement rethrows the current exception at least on MSVC

Reply:
And on GNU / clang it seems, feel free to merge

Reply:
It reuses the existing object, so it should be equivalent to your code snippet